### PR TITLE
Move most code (except CLI) into lib/

### PR DESCRIPTION
**Pull request by foolip**
The idea is to make lib/validate.js more testable, by splitting it
into smaller parts while keeping the top-level validate.js as-is.

That config.json is loaded directly from so many places is a bit
unfortunate, but left untouched for now.

### CODE REVIEW

**Comment by foolip** [comment]
@dontcallmedom I'm not entirely sure of the benefits of moving validate.js into lib/ in its current state, since it's very hard to test anyway. If you'd prefer, I could leave all of it in the root and move bits of it into lib/ as they're tested. I really just want to get to the bits about branch protection and test that code when I write it.

**Comment by dontcallmedom** [comment]
I'm fine with moving it to `lib/` - my only concern is losing the history of the code in the new `lib/validate.js`. I don't know if git supports history tracking when doing `git mv` and then creating a new file at the old place - is that what you did here?

**Comment by dontcallmedom** [comment]
("losing" in a loose sense, obviously)

**Comment by foolip** [comment]
If I had just moved the file with some small changes then git’s rename detection would have handled this, but because the original file also remains the new file will at best look like a copy.

I think I’ll revert the move and instead move smaller chunks. History will be hard to follow for any individual piece moved however.

**Comment by foolip** [comment]
@dontcallmedom I've updated the PR now to leave validate.js alone.
